api/v2/checluster_types.go

```diff
@@ -0,0 +1,637 @@
+	Components CheClusterComponents `json:"components"`
+	// Che authentication and TLS configuration.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=3
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Networking"
+	Networking CheClusterSpecNetworking `json:"networking,omitempty"`
+	// Configuration of an alternative registry that stores Che images.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=4
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Container registry"
+	ContainerRegistry CheClusterContainerRegistry `json:"containerRegistry"`
+}
+
+// Development environments configuration.
+// +k8s:openapi-gen=true
+type CheClusterDevEnvironments struct {
+	// Workspaces persistent storage.
+	// +optional
+	Storage WorkspaceStorage `json:"storage"`
+	// Default plug-ins applied to Dev Workspaces.
+	// +optional
+	DefaultPlugins []WorkspaceDefaultPlugins `json:"defaultPlugins,omitempty"`
+	// The node selector that limits the nodes that can run the workspace pods.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// The pod tolerations put on the workspace pods to limit where the workspace pods can run.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// User's default namespace.
+	// +optional
+	DefaultNamespace DefaultNamespace `json:"defaultNamespace,omitempty"`
+	// Trusted certificate settings.
+	// +optional
+	TrustedCerts TrustedCerts `json:"trustedCerts,omitempty"`
+}
+
+// Che components configuration.
+// +k8s:openapi-gen=true
+type CheClusterComponents struct {
+	// DevWorkspace operator configuration.
+	// +optional
+	DevWorkspace DevWorkspace `json:"devWorkspace"`
+	// General configuration settings related to the Che server.
+	// +optional
+	CheServer CheServer `json:"cheServer"`
+	// Configuration settings related to the Plugin registry used by the Che installation.
+	// +optional
+	PluginRegistry PluginRegistry `json:"pluginRegistry"`
+	// Configuration settings related to the Devfile registry used by the Che installation.
+	// +optional
+	DevfileRegistry DevfileRegistry `json:"devfileRegistry"`
+	// Configuration settings related to the database used by the Che installation.
+	// +optional
+	Database Database `json:"database"`
+	// Configuration settings related to the Dashboard sed by the Che installation.
+	// +optional
+	Dashboard Dashboard `json:"dashboard"`
+	// Kubernetes Image Puller configuration.
+	// +optional
+	ImagePuller ImagePuller `json:"imagePuller"`
+	// Che server metrics configuration.
+	// +optional
+	Metrics ServerMetrics `json:"metrics"`
+}
+
+// Configuration settings related to the Networking used by the Che installation.
+// +k8s:openapi-gen=true
+type CheClusterSpecNetworking struct {
+	// Comma separated list of labels that can be used to organize and categorize objects by scoping and selecting.
```

Comment:
Doc looks to be out of date -- says "comma separated" but field is `map[string]string`